darshan-util/pydarshan/tests/test_heatmap_handling.py

```diff
@@ -0,0 +1,575 @@
+    # `tests/input/sample-dxt-simple.darshan`, which only has write data
+    "ops, expected_df_dict",
+    [
+        (
+            # check the result using both operations
+            ["read", "write"],
+            {
+                "DXT_POSIX": {
+                    "read": pd.DataFrame(),
+                    "write": pd.DataFrame(
+                        columns=["length", "start_time", "end_time", "rank"],
+                        data=np.array(
+                            [
+                                [40, 0.10337884305045009, 0.10338771319948137, 0],
+                                [4000, 0.10421665315516293, 0.10423145908862352, 0],
+                            ]
+                        ),
+                    ),
+                },
+            },
+        ),
+        (
+            # check the result for only the "read" operation, should be empty
+            ["read"],
+            {"DXT_POSIX": {"read": pd.DataFrame()}},
+        ),
+        (
+            # the results for only checking the "write" data should be the same
+            # as checking both operations
+            ["write"],
+            {
+                "DXT_POSIX": {
+                    "write": pd.DataFrame(
+                        columns=["length", "start_time", "end_time", "rank"],
+                        data=np.array(
+                            [
+                                [40, 0.10337884305045009, 0.10338771319948137, 0],
+                                [4000, 0.10421665315516293, 0.10423145908862352, 0],
+                            ]
+                        ),
+                    ),
+                },
+            },
+        ),
+    ],
+)
+def test_get_single_df_dict(expected_df_dict, ops):
+    # regression test for `heatmap_handling.get_single_df_dict()`
+
+    report = darshan.DarshanReport("tests/input/sample-dxt-simple.darshan")
+
+    actual_df_dict = heatmap_handling.get_single_df_dict(
+        report=report, mods=["DXT_POSIX"], ops=ops
+    )
+
+    # make sure we get the same key(s) ("DXT_POSIX")
+    assert actual_df_dict.keys() == expected_df_dict.keys()
+    # make sure we get the correct key(s)
+    assert actual_df_dict["DXT_POSIX"].keys() == expected_df_dict["DXT_POSIX"].keys()
+    # also check that we only get the key(s) we requested
+    assert list(actual_df_dict["DXT_POSIX"].keys()) == ops
+
+    if "read" in ops:
+        # for the read case, check that we get an empty dataframe
+        assert actual_df_dict["DXT_POSIX"]["read"].empty
+
+    if "write" in ops:
+        # check that we get the same column names
+        assert_array_equal(
+            actual_df_dict["DXT_POSIX"]["write"].columns,
+            expected_df_dict["DXT_POSIX"]["write"].columns,
+        )
+
+        # verify the returned values are the same
+        assert_array_equal(
+            actual_df_dict["DXT_POSIX"]["write"].values,
+            expected_df_dict["DXT_POSIX"]["write"].values,
```

Comment:
these are floats being compared--I think I already added a comment about using `assert_allclose()` in these cases, so please do that in a follow-up when I'm done here